lib/mongoid/copyable.rb

```diff
@@ -19,29 +19,49 @@ module Copyable
     def clone
       # @note This next line is here to address #2704, even though having an
       # _id and id field in the document would cause problems with Mongoid
-      # elsewhere.
+      # elsewhere. Note this is only done on the root document as we want
+      # to maintian the same _id on the embedded documents.
       attrs = clone_document.except(*self.class.id_fields)
+      clone_with_hash(self.class, attrs)
+    end
+    alias :dup :clone
+
+    private
+
+    # Create clone of a document of the given klass with the given attributes
+    # hash. This is used recursively so that embedded associations are cloned
+    # safely.
+    #
+    # @param klass [ Class ] The class of the document to create.
+    # @param attrs [ Hash ] The hash of the attributes.
+    #
+    # @return [ Document ] The new document.
+    def clone_with_hash(klass, attrs)
```

Comment:
If this method doesn't reference any instance variables can it be a class method?

Reply:
Sure. I have changed it, let me know if this it what you envisioned.